docs/sdkx-react-native/delegates.mdx

```diff
@@ -0,0 +1,227 @@
+
+```
+    helpshiftEmitter.addListener('onEventOccurred', (data) => {
+      if (data.eventName == 'messageAdd') {
+        //Your code here.
+        //Event data keys: 
+        //  -` type` or `attachment` or `text`
+        //  - `body`
+      }
+    });  
+```
+
+### CSAT Submit Event {#csat-submit-event}
+
+This event is triggered when the user submits a CSAT rating after the conversation ends. The event data object has `rating` and `additionalFeedback` keys, which indicates the (star) rating and the additional comments provided by the user with the CSAT form. For your reference, see the below example.
+
+```
+    helpshiftEmitter.addListener('onEventOccurred', (data) => {
+      if (data.eventName == 'csatSubmit') {
+        //Your code here.
+      }
+    });  
+```
+
+### Conversation End Event {#conversation-end-event}
+
+This event is triggered when the conversation ends (resolved or rejected) and it cannot be reopened.
+
+```
+    helpshiftEmitter.addListener('onEventOccurred', (data) => {
+      if (data.eventName == 'conversationEnd') {
+        //Your code here.
+      }
+    });  
+```
+
+### Conversation Rejected Event {#conversation-rejected-event}
+
+This event is triggered when an agent rejects the conversation.
+
+```
+    helpshiftEmitter.addListener('onEventOccurred', (data) => {
+      if (data.eventName == 'conversationRejected') {
+        //Your code here.
+      }
+    });  
+```
+
+### Conversation Resolved Event {#conversation-resolved-event}
+
+This event is triggered when an agent resolves the conversation.
+
+```
+    helpshiftEmitter.addListener('onEventOccurred', (data) => {
+      if (data.eventName == 'conversationResolved') {
+        //Your code here.
+      }
+    });  
+```
+
+### Conversation Reopened Event {#conversation-reopened-event}
+
+When the resolution question is enabled, the users are asked if they're satisfied with the resolution. If the user rejects it and sends a new message, then the conversation is reopened and the Conversation Reopened event is triggered.
+
+```
+    helpshiftEmitter.addListener('onEventOccurred', (data) => {
+      if (data.eventName == 'conversationReopened') {
+        //Your code here.
+      }
+    });  
+```
+
+### User Authentication Failed Event {#user-authentication-failed-event}
+
+
+If you have `user authentication feature` enabled on the Dashboard and if you pass an invalid token in the `Login(userDataMap)`, then you will receive this event with reason string. Check [here for more info](/sdkx-react-native/users)
+
+You have to use `BindAuthFailureDelegate` to listen to Authentication failure events.
+
+The authentication failure reason is wrapped in an enum `EHelpshiftAuthenticationFailureReason` Example:
+
+Reason type :
+
+- `EHelpshiftAuthenticationFailureReason.REASON_INVALID_AUTH_TOKEN`
```

Comment:
All this is Unreal docs i think, please check again

Reply:
Updated